src/cone_detector.cpp

```diff
@@ -133,167 +285,160 @@ bool ConeDetector::hullIsValid(std::vector<cv::Point>& hull) {
+        top_left_x, top_right_x, bottom_left_x, bottom_right_x, min_y, max_y, length_top, length_bottom, height, aspect_ratio);
     if (aspect_ratio > max_aspect_ratio_) {
-        ss << ", REJECT, aspect_ration greater than: " << std::setprecision(4) << max_aspect_ratio_;
-        ROS_INFO("%s", ss.str().c_str());
+        ss << " REJECT, aspect_ratio: " << std::setprecision(4) << aspect_ratio << " greater than max: " << std::setprecision(4) << max_aspect_ratio_;
+        ROS_INFO_COND_NAMED("debug_", "cone_detector", "] %s", ss.str().c_str());
         return false;
     }
 
-   if (length_top < 0) ss << ", REJECT bad tlen";
-    if (length_bottom <= 0) ss << ", REJECT bad blen";
-    if (length_top >= (0.75 * length_bottom)) ss << ", REJECT bad shape";
-    ROS_INFO("%s", ss.str().c_str());
-    bool result = (length_top >= 0) &&
-                  (length_bottom > 0) &&
+    bool result = (length_bottom > 0) &&
                   (length_top < (0.75 * length_bottom));
+    if (result) {
+        ss << " ACCEPTED";
+    } else {
+        if (length_bottom <= 0) ss << "REJECT zero bottom length. ";
+        if (length_top >= (0.75 * length_bottom)) ss << "REJECT top longer that 0.75 of bottom. ";
+    }
+    
+    ROS_INFO_COND_NAMED(debug_, "cone_detector", " %s", ss.str().c_str());
     return result;
 }
 
 void ConeDetector::imageCb(cv::Mat& original_image) {
-    static const cv::Scalar color = cv::Scalar(0, 0, 255);
-    static const cv::Scalar non_primary_color = cv::Scalar(0, 255, 255);
-
-    clock_t start;
-    double duration_resize = 0;         // Time taken for resize call.
-    double duration_cvtColor = 0;       // Time taken for cvtColor call.
-    double duration_erode_dilate_ = 0;  // Time taken for erode/dilate calls.
-    double duration_inRange = 0;        // Time taken for inRange call.
-    double duration_copyTo = 0;         // Time taken for copyTo operation.
-    double d_theta = 0;    // Time taken to find largets blob.
-    double duration_imshow = 0;         // Time taken for imshow calls
-    double duration_findContours = 0;   // Time taken for findContours call.
-    double duration_find_largest = 0;   // Time taken to find the largest contour.
-
-    cv::Mat img_HSV;
-    if (show_step_times_) start = clock();
-    cv::Size resize_dimensions(resize_x_, resize_y_);
+    original_image.copyTo(last_image_); // Save so kmeans has an image to work on.
+    last_image_count_++;    // Count so kmeans knows that at least one image has been received.
+    static const cv::Scalar color = cv::Scalar(0, 0, 255);  // Color of annotated primary circle.
+    static const cv::Scalar non_primary_color = cv::Scalar(0, 255, 255);    // Color of annotated non-primary circles.
+
+    cv::Mat img_hsv;    // We'll work in HSV space for color detction.
+    cv::Size resize_dimensions(resize_x_, resize_y_);   // Down sample size for faster computation.
```

Comment:
It's changed by configuration. Right now there is only a single use, so I don't see an advantage or disadvantage in making it a member variable.